hydro_lang/src/ir.rs

```diff
@@ -49,9 +51,180 @@ impl ToTokens for DebugExpr {
+                // Look for calls to stageleft::runtime_support::fn_*
+                if self.is_stageleft_runtime_support_call(&path_expr.path) {
+                    // Try to extract the closure from the arguments
+                    if let Some(closure) = self.extract_closure_from_args(&call.args) {
+                        *self.simplified_result.borrow_mut() = Some(format!("q!({})", closure));
+                        return;
+                    }
+                }
+            }
+        }
+
+        // Continue visiting child expressions using the default implementation
+        // Use the default visitor to avoid infinite recursion
+        syn::visit_mut::visit_expr_mut(self, expr);
+    }
+}
+
+impl QMacroSimplifier {
+    fn is_stageleft_runtime_support_call(&self, path: &syn::Path) -> bool {
+        // Check if this is a call to stageleft::runtime_support::fn_*
+        if let Some(last_segment) = path.segments.last() {
+            if last_segment.ident.to_string().starts_with("fn_") {
+                // Check if the path contains stageleft and runtime_support
+                let path_str = path
+                    .segments
+                    .iter()
+                    .map(|seg| seg.ident.to_string())
+                    .collect::<Vec<_>>()
+                    .join("::");
+                return path_str.contains("stageleft") && path_str.contains("runtime_support");
+            }
+        }
+        false
+    }
+
+    fn extract_closure_from_args(
+        &self,
+        args: &syn::punctuated::Punctuated<syn::Expr, syn::Token![,]>,
+    ) -> Option<String> {
+        // Look through the arguments for a closure expression
+        for arg in args {
+            if let syn::Expr::Closure(closure) = arg {
+                // Format the closure nicely
+                let closure_str = closure.to_token_stream().to_string();
+                return Some(self.cleanup_closure_content(&closure_str));
+            }
+
+            // Also check for closures nested in other expressions (like blocks)
+            if let Some(closure_str) = self.find_closure_in_expr(arg) {
+                return Some(self.cleanup_closure_content(&closure_str));
+            }
+        }
+        None
+    }
+
+    fn find_closure_in_expr(&self, expr: &syn::Expr) -> Option<String> {
+        match expr {
+            syn::Expr::Block(block) => {
+                // Look for closures in block statements
+                for stmt in &block.block.stmts {
+                    if let syn::Stmt::Expr(syn::Expr::Closure(closure), _) = stmt {
+                        return Some(closure.to_token_stream().to_string());
+                    }
+                }
+            }
+            syn::Expr::Closure(closure) => {
+                return Some(closure.to_token_stream().to_string());
+            }
+            _ => {}
+        }
+        None
+    }
+
+    fn cleanup_closure_content(&self, content: &str) -> String {
```

Comment:
Should not be needed with `prettyplease`?

Reply:
I prettypleased it now.